controllers/managedfusion_controller.go

```diff
@@ -231,6 +236,13 @@ func (r *ManagedFusionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("%s.smtp_config is not in a valid yaml format: %v", managedFusionSecretName, err)
 	}
 
+	if config, found := r.managedFusionSecret.Data["remote_write_config"]; found {
+		r.remoteWriteConfigData = &remoteWriteConfigData{}
+		if err := yaml.Unmarshal(config, r.remoteWriteConfigData); err != nil {
+			return ctrl.Result{}, fmt.Errorf("%s.remote_write_config is not in a valid yaml format: %v", managedFusionSecretName, err)
```

Comment:
Errors are not log messages and should be structured accordingly. And we should not assume that the payload is invalid just because unmarshal didn't work.
```suggestion
			return ctrl.Result{}, fmt.Errorf("Failed to unmarshal agent remote_write_config at %s,  %v", managedFusionSecretName, err)
```

And please apply the same for the equivalent errors above this one.

Reply:
Done

Reply:
@rohan47 I see only a change for the message you added. The ask above was:
```And please apply the same for the equivalent errors above this one.```

Please fix